src/components/transport_manager/test/transport_manager_default_test.cc

```diff
@@ -29,45 +29,71 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
+
 #include "gtest/gtest.h"
 #include "transport_manager/transport_manager.h"
 #include "transport_manager/transport_manager_default.h"
-#include "resumption/last_state.h"
 #include "transport_manager/mock_transport_manager_settings.h"
-#include "resumption/last_state.h"
+#include "resumption/mock_last_state.h"
 
 namespace test {
 namespace components {
 namespace transport_manager_test {
 
+using ::resumption::MockLastState;
 using ::testing::Return;
+using ::testing::ReturnRef;
+using ::testing::NiceMock;
 
 TEST(TestTransportManagerDefault, Init_LastStateNotUsed) {
   MockTransportManagerSettings transport_manager_settings;
   transport_manager::TransportManagerDefault transport_manager(
       transport_manager_settings);
-  resumption::LastState last_state("app_storage_folder", "app_info_storage2");
+
+  NiceMock<resumption::MockLastState> mock_last_state;
+  Json::Value custom_dictionary = Json::Value();
+
+  ON_CALL(mock_last_state, get_dictionary())
+      .WillByDefault(ReturnRef(custom_dictionary));
 
   EXPECT_CALL(transport_manager_settings, use_last_state())
       .WillRepeatedly(Return(false));
   EXPECT_CALL(transport_manager_settings, transport_manager_tcp_adapter_port())
       .WillRepeatedly(Return(1u));
-  transport_manager.Init(last_state);
+
+  transport_manager.Init(mock_last_state);
+  transport_manager.Stop();
 }
 
-TEST(TestTransportManagerDefault, DISABLED_Init_LastStateUsed) {
-  // TODO (dcherniev) : investigate and fix SegFault issue
+TEST(TestTransportManagerDefault, Init_LastStateUsed) {
   MockTransportManagerSettings transport_manager_settings;
   transport_manager::TransportManagerDefault transport_manager(
       transport_manager_settings);
-  resumption::LastState last_state("app_storage_folder", "app_info_storage2");
+
+  NiceMock<resumption::MockLastState> mock_last_state;
+  Json::Value custom_dictionary;
+  Json::Value tcp_device;
+  tcp_device["name"] = "unique_tcp_device_name";
+  tcp_device["address"] = "57.48.0.1";
+  tcp_device["applications"][0]["port"] = "12345";
+  Json::Value bluetooth_device;
+  bluetooth_device["name"] = "unique_bluetooth_device_name";
+  bluetooth_device["address"] = "57.48.0.2";
+  bluetooth_device["applications"][0]["port"] = "12345";
+  custom_dictionary["TransportManager"]["TcpAdapter"]["devices"][0] =
+      tcp_device;
+  custom_dictionary["TransportManager"]["BluetoothAdapter"]["devices"][0] =
+      bluetooth_device;
```

Comment:
@dcherniev, may you move map keys (like `"name"`, `"address"`, etc) to a constant variables and then use them?
Same thing for `"port"` value.
## 
Please check in other files too.